deepcell/image_generators/semantic.py

```diff
@@ -105,8 +105,16 @@ def __init__(self,
                              'should have rank 4. You passed an array '
                              'with shape', X.shape)
 
-        self.x = np.asarray(X, dtype=K.floatx())
-        self.y = np.asarray(y, dtype='int32')
+        if X.dtype in ['float32', 'int32', 'float64', 'int64']:
+            logging.warning('X data dtype is {}: this will increase memory use during '
+                            'preprocessing. Consider using a smaller dtype'.format(X.dtype))
+
+        if y.dtype in ['float32', 'int32', 'float64', 'int64']:
+            logging.warning('y data dtype is {}: this will increase memory use during '
+                            'preprocessing. Consider using a smaller dtype.'.format(y.dtype))
+
+        self.x = np.asarray(X, dtype='float16')
+        self.y = np.asarray(y, dtype='int16')
```

Comment:
It looks like we warn the user and then throw away the dtype we warn them about. Maybe we should update the warning message to tell them what dtype it will be cast to?

Reply:
It turns out that the X and y that are passed in also take up significant memory. So either way, we convert to float16, but that's not enough to get rid fo the memory cost of loading them in the first place if there were float64, for example